plugins/available/goenv.plugin.bash

```diff
@@ -0,0 +1,29 @@
+cite about-plugin
+about-plugin 'Init goenv, if installed. Plays nicely with package managers'
+
+# GOENV_ROOT does not need to be configured.
+# If it is configured, it should not be overridden.
+# However, let's try to handle the case where:
+# - goenv is not already on the path
+# - GOENV_ROOT is not configured
+# - User has installed goenv in ~/.goenv
+# In this case, we'll do the user a solid and configure
+# the goenv they've taken the time to install.
+if ! _command_exists goenv && [ -z "${GOENV_ROOT}" ] && [ -x "${HOME}/.goenv/bin/goenv" ]; then
+	export GOENV_ROOT="${HOME}/.goenv"
+fi
```

Comment:
I *strongly* disagree with your premise. I rely on the *env plugins setting *_ROOT.

Please do not deviate from the pattern used in our other plugins. Just as I suggest in my very first PR comment, if you want to make a larger change, please make that a separate PR.

Reply:
I don't see how you can disagree with this ...

Your PR explicitly overrides GOENV_ROOT to `~/.goenv` *whenever* it is not already set. 
 This *always* overrides goenv's built-in default handling.

My PR *only* overrides GOENV_ROOT to `~/.goenv` when it is not set **AND** when we have a strong indicator that the user *wants* to use it.  i.e the presence of `/.goenv/bin/goenv` gives us permission to explicitly set the value. When `~/.goenv/bin/goenv` is not present, we defer to goenv's built-in default behavior.

With that, could you elaborate on what you disagree with and how you perceive your PR as being different?

Reply:
> how you perceive your PR as being different

you already described how it's different:

> Your PR explicitly overrides GOENV_ROOT to ~/.goenv whenever it is not already set.
This always overrides goenv's built-in default handling.

that's pretty much it after all the fixups

> I don't see how you can disagree with this ...

I think I say plainly that I use `*_ROOT` variables. Admittedly, I think I only have scripts that capitalize on `PYENV_ROOT`, but again, I state very plainly that [every](https://github.com/Bash-it/bash-it/blob/master/plugins/available/nodenv.plugin.bash) [other](https://github.com/Bash-it/bash-it/blob/master/plugins/available/pyenv.plugin.bash) [env](https://github.com/Bash-it/bash-it/blob/master/plugins/available/pyenv.plugin.bash) plugin has set this standard. If you want to deviate on some purist viewpoint, by all means, please do that in a different PR,

Reply:
I reject the premise of an `"accepted pattern"` on this subject.  the `pyenv` plugin doesn't even conditionally set the variable, IT JUST SETS IT!  Should that be an accepted pattern? NO, it shouldn't.

I will accept, however, that even if the tools that the plugin are meant to support do NOT require the variables to function, users might come to depend on them.

That is the case here.  `goenv` does not require the variable to be set.   In fact, the [Homebrew section](https://github.com/syndbg/goenv/blob/master/INSTALL.md#homebrew-on-mac-os-x) of the install document doesn't even mention the variable.

The REAL pattern *should* be for the tools to ensure that the variable gets set as part of their `eval` if it is not already set when called.

All this is to say : I see the value you get from trading an undesired hard-coded value for a guarantee that the variable is set when the plugin exist.

I will update this PR with that idea in mind.   Thank you,

-DF

Reply:
> I reject the premise of an "accepted pattern" on this subject.

As you've already said you don't use this tool, so I appreciate your viewpoint (I partially agree) but you have no skin in this game. Why be such a stalwart? You could just as easily do all of this in a follow-up PR.

Reply:
Hello again,

I'm a little confused as to why you are harping on the first sentence of that post, but not acknowledging the last sentences in that same post:

> All this is to say : I see the value you get from trading an undesired hard-coded value for a guarantee that the variable is set when the plugin exits.
>
> I will update this PR with that idea in mind. Thank you,

I also don't know if you're reading the actual changes in the PR, specifically the updates made as per this discussion:
```
# Use a reasonable default to ensure variable is always present
...
GOENV_ROOT="${GOENV_ROOT:-${HOME}/.goenv}"
```